### ISSUE TITLE
Docker container with Prometheus 2.29.0 not running on a Raspberry Pi

### ISSUE THREAD

**Issue by sebastianbrosch** [prometheus/prometheus#9190]
**What did you do?**

I tried to upgrade Prometheus from version 2.28.1 to version 2.29.0 on a Docker stack on a Raspberry Pi. I also tried to run Prometheus with Docker directly (no stack, no docker-compose) using [this command (just changed IP and port)](https://github.com/prometheus/prometheus#docker-images):

running:
```
docker run --name prometheus -d -p 127.0.0.1:9090:9090 prom/prometheus:v2.28.1
```
not running (Status: Exited (2) - after a few seconds):
```
docker run --name prometheus -d -p 127.0.0.1:9090:9090 prom/prometheus:v2.29.0
```

**What did you expect to see?**

I expected a successful upgrade to version 2.29.0 (Docker stack) or a running container using this command:

```
docker run --name prometheus -d -p 127.0.0.1:9090:9090 prom/prometheus:v2.29.0
```

**What did you see instead? Under which circumstances?**

 - The container of the stack could not be build successfully. 
 - The container is not running after `docker run` command (Status: Exited (2)).

This error message is shown in `docker logs` of the exited container:

![image](https://user-images.githubusercontent.com/6590066/128999357-7c28cc02-4f37-4dbd-830c-ec9fe1262012.png)

**Environment**

* System information:

Output of `uname -srm` on a Raspberry Pi 4:

```
Linux 5.10.17-v7l+ armv7l
```

Output of `docker version`

```
Client: Docker Engine - Community
 Version:           20.10.6
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        370c289
 Built:             Fri Apr  9 22:46:18 2021
 OS/Arch:           linux/arm
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.6
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       8728dd2
  Built:            Fri Apr  9 22:44:17 2021
  OS/Arch:          linux/arm
  Experimental:     false
 containerd:
  Version:          1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Comment by roidelapluie** [comment]
Sounds like:

https://github.com/golang/go/issues/36606
https://pkg.go.dev/sync/atomic#pkg-note-BUG

**Comment by roidelapluie** [comment]
cc @codesome

**Comment by roidelapluie** [comment]
I have open https://github.com/prometheus/prometheus/pull/9192 to fix this issue and we will release a bugfix shortly.

**Comment by roidelapluie** [comment]
Closed in #9192

**Comment by sebastianbrosch** [comment]
@roidelapluie I tested the Docker command and upgrade on Docker stack. Both are running now with v2.29.1. Thank you!
